lib/probes/http.js

```diff
@@ -570,13 +570,13 @@ function patchServer (module, options, protocol) {
           error: res.statusCode >= 500 && res.statusCode <= 599
         }
 
-        // it replies with the txname that was actually used if there was an error
+        // the return value can be an integer code if there was an error. if no
+        // error then it will be the txname that was used, which might not match
+        // the txname the user specified if it wasn't valid.
         const txname = ao.reporter.sendHttpSpan(args);
 
-        // TODO BAM the txname not null check can be removed once appoptics-bindings has been
-        // changed to return the integer error code instead of a null string (version > 7.1.0)
-        if (typeof txname === 'string' && txname) {
-          // if there is a txname and it doesn't match
+        if (typeof txname === 'string') {
+          // see if the name used doesn't match the name provided
```

Comment:
on line 584 should it be `txname.endsWith(args.txname)`

Reply:
yes, it should. thanks.

i think even more directly i can get rid of the domainPrefix check because the .endsWith will ignore that.
so: `if (!txname.endsWith(args.txname)) {` seems to work.